src/node/miner.cpp

```diff
@@ -90,18 +90,17 @@ void ApplyArgsManOptions(const ArgsManager& args, BlockAssembler::Options& optio
     if (const auto blockmintxfee{args.GetArg("-blockmintxfee")}) {
         if (const auto parsed{ParseMoney(*blockmintxfee)}) options.blockMinFeeRate = CFeeRate{*parsed};
     }
+
     options.print_modified_fee = args.GetBoolArg("-printpriority", options.print_modified_fee);
+    options.coinbase_max_additional_weight = args.GetIntArg("-maxcoinbaseweight", options.coinbase_max_additional_weight);
 }
 
 void BlockAssembler::resetBlock()
 {
     inBlock.clear();
-
-    // Reserve space for coinbase tx
-    nBlockWeight = m_options.coinbase_max_additional_weight;
-    nBlockSigOpsCost = m_options.coinbase_output_max_additional_sigops;
-
     // These counters do not include coinbase tx
+    nBlockWeight = 0;
+    nBlockSigOpsCost = 0;
```

Comment:
This changes the meaning of `getmininginfo` fields unexpectedly, and forces us to add in the coinbase reserved size over and over. Seems like it would be better to include it here once.

Reply:
I agree we should not change the `weight` and `sigops` fields in `getblocktemplate` and `getmininginfo`.

We could either adjust the value in the RPC code, or as @luke-jr suggests restore the original behavior here.

Additionally we should clarify the help text for these RPC methods that they include the coinbase reserved weight.

Reply:
> This changes the meaning of `getmininginfo` fields unexpectedly, and forces us to add in the coinbase reserved size over and over. Seems like it would be better to include it here once.

> We could either adjust the value in the RPC code, or as @luke-jr suggests restore the original behavior here.

I've reverted and included it but clarify the `getmininginfo` help text.

> I agree we should not change the `weight`, `currentblockweight` and `sigops` fields in `getblocktemplate` and `getmininginfo`.

This only affects `getmininginfo` I think the `weight` and `sigops` results of `getblocktemplate` are for individual transactions in the block template.

Reply:
> I think the `weight` and `sigops` results of `getblocktemplate` are for individual transactions in the block template.

Ah that would make sense, since I tried to add those totals in an early version of #27433.